dist/src/test/java/io/camunda/zeebe/shared/security/SecurityConfigurationTest.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.shared.security;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.shared.security.SecurityConfigurationTest.TestCase.Builder;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.env.Environment;
+import org.springframework.core.type.MethodMetadata;
+
+public class SecurityConfigurationTest {
+
+  @Mock private Environment environment;
+
+  private ConditionContext conditionContext;
+  private MethodMetadata methodMetadata;
+  private final SecurityConfiguration.GatewaySecurityAuthenticationEnabledCondition condition =
+      new SecurityConfiguration.GatewaySecurityAuthenticationEnabledCondition();
```

Comment:
🔧 The class may be under test, but only the `Condition` interface is actually used in the test

```suggestion
  private final Condition condition =
      new SecurityConfiguration.GatewaySecurityAuthenticationEnabledCondition();
```